### ISSUE TITLE
Nix fails to install when an existing group is assigned the GID of 30000

### ISSUE THREAD

**Issue by dreemkiller** [NixOS/nix#3846]
**Describe the bug**

I am installing on an x86_64 Ubuntu system (/etc/issue contents: Ubuntu 18.04.3 LTS )
When I run the install script, I get the following message:
```

~~> Setting up the build group nixbld

---- sudo execution ------------------------------------------------------------
I am executing:
    $ sudo groupadd -g 30000 --system nixbld

Create the Nix build group, nixbld

groupadd: GID '30000' already exists

---- oh no! --------------------------------------------------------------------
```

**Steps To Reproduce**
```
sh nix_install.sh --daemon
```
"click through" the questions.

**Expected behavior**
I would expect NIX to have agility regarding the GIDs of the groups it needs to use. It does not appear to have that (or the install script itself is not agile enough to handle it)

**Additional context**

executing
```getent group 30000```
on my system returns the name of an existing group on the system. It would be unwise of me to change the GID of this group (This is an IT-managed system and I do not know what the existing group is for)

**Comment by Kloenk** [comment]
Nix itself can handle different groups, as set in the nix.conf. so I would guess that this error comes from the install script

**Comment by dreemkiller** [comment]
That makes sense, as section 4.5 of the manual (https://nixos.org/nix/manual/#sect-nix-install-binary-tarball) mentions the ability to change the GID in the ```install-multi-user``` file.

**Comment by Kloenk** [comment]
So can you just change that, and install it?

**Comment by dreemkiller** [comment]
Yes I can, but I would hope that the install script would be able to handle that case.

**Comment by zimbatm** [comment]
As a curiosity, what group do you have on GID 30000?

**Comment by dreemkiller** [comment]
Since it's a system managed by my IT department, all I know is the name (not its purpose), and I'm not sure if I should disclose it, so I won't.

**Comment by FRidh** [comment]
Reminds me when I was working at a large corporation, there I also had to modify the installer to increase the *ID's.

**Comment by Asqiir** [comment]
I might be late to the party, but I found a simple workaround, which makes it unnecessary to modify any script.

How:

1. Find a free `gid` (`getent group | grep <the-gid>`) should return nothing
2. run `export NIX_BUILD_GROUP_ID=<the-gid>`
3. Install nix as the [installation](https://nixos.org/download.html) page says (`sh <(curl -L https://nixos.org/nix/install) --daemon`)
4. maybe remove the permanently set variable again ;-)

Why: (links are linux-specific)

[The first script to be run](https://nixos.org/nix/install) downloads a [tar](https://releases.nixos.org/nix/nix-2.18.0/nix-2.18.0-x86_64-linux.tar.xz) which again contains a script called `install` which then calls another script from the same tar called `install-multi-user`.

This last script is the actual installer for multi-user on linux case. One of the first lines is

```bash
readonly NIX_BUILD_GROUP_ID="${NIX_BUILD_GROUP_ID:-30000}"
```

which sets the shell variable `NIX_BUILD_GROUP_ID` to `30000` if and only if it is still unset. Using a different value is as simple as overriding this variable.


How the script could be improved:

When the scripts each call the subsequent scripts, they hand over all command line parameters.

The last script (the one setting the hardcoded constant) could take a command line argument for overriding the value. A user can then set the variable in the initial call, which used to be `sh <(curl -L https://nixos.org/nix/install) --daemon` and is now `sh <(curl -L https://nixos.org/nix/install) --daemon --build-gid 30500`.

Instead of the current error message the user can get an explanation about this flag.

Also relevant for #6224, and partially so to #4915.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
